src/Spd.Resource.Repository/PersonLicApplication/PersonLicApplicationRepository.cs

```diff
@@ -139,6 +141,21 @@ public async Task<LicenceApplicationResp> GetLicenceApplicationAsync(Guid licenc
         return appResp;
     }
 
+    public async Task<LicenceApplicationCmdResp> UpdateSwlSoleProprietorApplicationAsync(Guid swlAppId, Guid bizLicAppId, CancellationToken ct)
+    {
+        var swlApp = await _context.spd_applications
+            .Where(a => a.spd_applicationid == swlAppId)
+            .SingleOrDefaultAsync(ct);
+        if (swlApp == null) throw new ApiException(HttpStatusCode.BadRequest, "cannot find the swl application.");
+
+        var bizLicApp = await _context.spd_applications.Where(a => a.spd_applicationid == bizLicAppId).SingleOrDefaultAsync(ct);
+        if (bizLicApp == null) throw new ApiException(HttpStatusCode.BadRequest, "cannot find the business application.");
+
+        _context.SetLink(swlApp, nameof(swlApp.spd_BusinessLicenseId), bizLicApp);
+        await _context.SaveChangesAsync(ct);
+        return new LicenceApplicationCmdResp((Guid)swlApp.spd_applicationid, swlApp._spd_applicantid_value, swlApp._spd_organizationid_value);
```

Comment:
why we do casting (Guid) when passing a guid into the function?
it means in this case spd_apliplicationId is nullable?

Reply:
because spd_applicationid is nullable, that's from dynamics